bql/semantic/hooks.go

```diff
@@ -664,6 +671,59 @@ func whereObjectClause() ElementHook {
 	return f
 }
 
+// whereFilterClause returns an element hook that updates the working filter clause and,
+// if the filter clause is complete, populates the filters list of the statement.
+func whereFilterClause() ElementHook {
+	var hook ElementHook
+	supportedFilterFunctions := map[string]bool{
+		"latest": true,
+	}
+
+	hook = func(st *Statement, ce ConsumedElement) (ElementHook, error) {
+		if ce.IsSymbol() {
+			return hook, nil
+		}
+
+		tkn := ce.Token()
+		currFilter := st.WorkingFilter()
+		switch tkn.Type {
+		case lexer.ItemFilterFunction:
+			if currFilter == nil {
+				return nil, fmt.Errorf("could not add filter function %q to nil filter clause", tkn.Text)
+			}
+			if currFilter.Operation != "" {
+				return nil, fmt.Errorf("invalid filter function %q on filter clause since already set to %q", tkn.Text, currFilter.Operation)
+			}
+			if !supportedFilterFunctions[tkn.Text] {
+				return nil, fmt.Errorf("filter function %q on filter clause is not supported", tkn.Text)
+			}
+			currFilter.Operation = tkn.Text
```

Comment:
WDYT of using st.WorkingFilter() instead of currFilter to be explicit that you are changing a external object instead of a local object?

Reply:
I have just refactored this filter hook to make it clearer.